src/concepts/plugins/frontend.md

```diff
@@ -0,0 +1,479 @@
+} as const;
+
+const getCount = (caido: Caido) => {
+  const storage = caido.storage.get() as PluginStorage | undefined;
+
+  if (storage) {
+    return storage.count;
+  }
+
+  return 0;
+}
+
+const increment = (caido: Caido) => {
+  const count = getCount(caido);
+  caido.storage.set({ count: count + 1 });
+}
+
+const decrement = (caido: Caido) => {
+  const count = getCount(caido);
+  caido.storage.set({ count: count - 1 });
+}
+
+const addPage = (caido: Caido) => {
+
+  const count = getCount(caido);
+
+  const body = document.createElement("div");
+  body.className = "my-plugin";
+  body.innerHTML = `
+    <div class="my-plugin__count">
+      <span>Count:</span>
+      <span class="my-plugin__value">${count}</span>
+    </div>
+    <div>
+      <button class="c-button" data-command="${Commands.increment}">Increment</button>
+      <button class="c-button" data-command="${Commands.decrement}">Decrement</button>
+    </div>
+  `;
+
+  const countElement = body.querySelector(".my-plugin__value") as HTMLElement;
+  const incrementButton = body.querySelector(`[data-command="${Commands.increment}"]`) as HTMLElement;
+  const decrementButton = body.querySelector(`[data-command="${Commands.decrement}"]`) as HTMLElement;
+
+  caido.storage.onChange((newStorage) => {
+    const storage = newStorage as PluginStorage | undefined;
+
+    if (storage) {
+      countElement.innerHTML = `${storage.count}`;
+      return;
+    }
+  });
+
+  incrementButton.addEventListener("click", () => {
+    increment(caido);
+  });
+
+  decrementButton.addEventListener("click", () => {
+    decrement(caido);
+  });
+
+  caido.navigation.addPage(Page, {
+    body,
+  });
+}
+
+
+export const init = (caido: Caido) => {
+
+  // Register commands
+  // Commands are registered with a unique identifier and a handler function
+  // The run function is called when the command is executed
+  // These commands can be registered in various places like command palette, context menu, etc.
+  caido.commands.register(Commands.increment, {
+    name: "Increment",
+    run: () => increment(caido),
+  });
+
+  caido.commands.register(Commands.decrement, {
+    name: "Decrement",
+    run: () => decrement(caido),
+  });
+
+  // Register command palette items
+  caido.commandPalette.register(Commands.increment);
+  caido.commandPalette.register(Commands.decrement);
+
+  // Register page
+  addPage(caido);
+
+  // Register sidebar
+  caido.sidebar.registerItem("My plugin", Page, {
+    icon: "fas fa-rocket",
+  });
+}
+```
+
+- In the external type definition file is `export { API as Caido } from “./types”;`. Now `Caido` is used as the interface to the API and is imported by `import type { Caido } from "@caido/sdk-frontend";`
+- Access to plugin storage is achieved by `import type { PluginStorage } from "./types";`.
+- The `Page` variable stores the path for the plugin page. When a user selects the plugin or navigates to `/my-plugin` - the page will be displayed.
```

Comment:
This is just a constant since it is a good programming practice, but not necessary in a beginner example